core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/sequenceMessage/SequenceMessageInvokedOperationExchangeItems.java

```diff
@@ -0,0 +1,92 @@
+ *******************************************************************************/
+package org.polarsys.capella.core.data.interaction.validation.sequenceMessage;
+
+import java.util.Collection;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.validation.AbstractModelConstraint;
+import org.eclipse.emf.validation.EMFEventType;
+import org.eclipse.emf.validation.IValidationContext;
+import org.polarsys.capella.common.data.modellingcore.AbstractExchangeItem;
+import org.polarsys.capella.core.data.cs.ExchangeItemAllocation;
+import org.polarsys.capella.core.data.fa.ComponentExchange;
+import org.polarsys.capella.core.data.fa.FunctionalExchange;
+import org.polarsys.capella.core.data.information.AbstractEventOperation;
+import org.polarsys.capella.core.data.information.Operation;
+import org.polarsys.capella.core.data.information.Service;
+import org.polarsys.capella.core.data.interaction.MessageKind;
+import org.polarsys.capella.core.data.interaction.SequenceMessage;
+import org.polarsys.capella.core.data.interaction.properties.dialogs.sequenceMessage.model.communications.InterfaceCommunication;
+import org.polarsys.capella.core.data.oa.CommunicationMean;
+import org.polarsys.capella.core.data.oa.OperationalActivity;
+import org.polarsys.capella.core.model.helpers.SequenceMessageExt;
+
+/**
+ */
+public class SequenceMessageInvokedOperationExchangeItems extends AbstractModelConstraint {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IStatus validate(IValidationContext ctx) {
+        //
+        // Preconditions.
+        //
+        EMFEventType eType = ctx.getEventType();
+        if (EMFEventType.NULL != eType) {
+            return ctx.createSuccessStatus();
+        }
+        EObject eObj = ctx.getTarget();
+        if (!(eObj instanceof SequenceMessage)) {
+            return ctx.createSuccessStatus();
+        }
+        SequenceMessage sequenceMessage = (SequenceMessage) eObj;
+        // Ignore TIMER SequenceMessage (by construction, they don't have a invoked operation).
+        if (MessageKind.TIMER == sequenceMessage.getKind()) {
+            return ctx.createSuccessStatus();
+        }
+        AbstractEventOperation invokedOperation = sequenceMessage.getInvokedOperation();
+        if (null != invokedOperation) {
+            Collection<AbstractExchangeItem> lstInvokedEI = SequenceMessageExt.getExchangeItemsFromOperation(sequenceMessage);
+            if (!sequenceMessage.getExchangedItems().containsAll(lstInvokedEI)) {
+                String invokedOperationDisplayName = computeDisplayNameForOperation(invokedOperation);
+                return ctx.createFailureStatus(sequenceMessage.getName(), invokedOperation.getName(), invokedOperationDisplayName);
+            }
+        }
+        return ctx.createSuccessStatus();
+    }
+    
+    private String computeDisplayNameForOperation(AbstractEventOperation invokedOperation) {
```

Comment:
Please use EObjectLabelProviderHelper.getMetaclassLabel were possible, instead of hardcoded values - it returns the properties defined in https://github.com/eclipse/capella/blob/master/core/plugins/org.polarsys.capella.core.data.gen.edit/plugin.properties

Reply:
Yes sure, you are right